src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -169,7 +169,7 @@ private synchronized void syncShardLeases(IKinesisProxy kinesisProxy,
 
         List<Shard> shards;
         if(CollectionUtils.isNullOrEmpty(latestShards)) {
-            shards = getShardListAtInitialPosition(kinesisProxy, initialPosition);
+            shards = isLeaseTableEmpty ? getShardListAtInitialPosition(kinesisProxy, initialPosition) : getCompleteShardList(kinesisProxy);
         } else {
```

Comment:
How did we catch this issue now? and how our test canary could have caught this? basically this should have prevented new leases from discovering right?

Reply:
I caught it since I saw in the logs that there were a lot of cache misses, basically the canary unblocks itself after a minute or two when it sees too many cache misses and forces a cache refresh.